File: aspnetapp/WINGS/Models/Operation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WINGS.Models
{
  public enum TmtcTarget
  {
    TmtcIf,
    Infostellar
  }

  public class Operation
  {
    [Column(TypeName = "varchar(255)")]
    public string Id { get; set; }

    [Column(TypeName = "varchar(64)")]
    public string PathNumber { get; set; }

    [Column(TypeName = "longtext")]
    public string Comment { get; set; }

    [Column(TypeName = "datetime(0)")]
    public DateTime CreatedAt { get; set; }
    
    public bool IsRunning { get; set; }
    
    public bool IsTmtcConnected { get; set; }

    [Column(TypeName = "varchar(64)")]
    public TlmCmdFileLocation FileLocation { get; set; }

    [Column(TypeName = "varchar(64)")]
    public TmtcTarget TmtcTarget { get; set; }

    [Required]
    public string ComponentId { get; set; }
    public Component Component { get; set; }
  }
}

File: aspnetapp/WINGS/ClientApp/src/models/Operation.ts
import { Component } from './Component';

export type TlmCmdConfigurationInfo = {
  tlmApid: string,
  cmdApid: string,
  compoName: string
};

export type Operation = {
  id: string,
  pathNumber: string,
  comment: string,
  isRunning: boolean,
  isTmtcConncted: boolean,
  fileLocation: string,
  tmtcTarget: string | null,
  operatorId: string | null,
  operator: {
    id: string,
    userName: string,
    role: string
  },
  componentId: string,
  component: Component,
  createdAt: string,
  satelliteId: string | null,
  planId: string | null,
  tlmCmdConfig: TlmCmdConfigurationInfo[]
}
